### ISSUE TITLE
[feat] `uv config` command

### ISSUE THREAD

**Issue by baggiponte** [astral-sh/uv#6042]
Now that `uv` is XDG compliant it might matter less; but I'd argue it'd be helpful to have a `uv config` namespace to access uv configurations:

```
# list all configurations
uv config

# see current value for foo
uv config foo

# set value
uv config foo bar

# alternatively
uv config get foo
uv config set foo bar
```

**Comment by zanieb** [comment]
Yeah we're interested in doing this eventually.

**Comment by kwaegel** [comment]
I'm also interested in this. Especially with the addition of multiple packages indexes, it would be really helpful to know what configuration values `uv` is picking up and where they're coming from (e.g. `uv config list`).

**Comment by CharString** [comment]
This feature would make issues like https://github.com/tox-dev/tox-uv/issues/253 much easier to fix. Projects down stream wouldn't have to reimplement the environment and configuration file parsing and merging.

**Comment by nonZero** [comment]
This is getting more important as people want to verify their `exclude-newer` settings works.

**Comment by beaugunderson** [comment]
> This is getting more important as people want to verify their `exclude-newer` settings works.

yes, as well as the ability to push this change out to all of our employees with an MDM script without worrying about either creating or editing an existing global `uv.toml`

for anyone looking at shoring that up after today's npm supply-chain shenanigans:

```bash
python3 -c "
import pathlib, re
p = pathlib.Path('~/.config/uv/uv.toml').expanduser()
p.parent.mkdir(parents=True, exist_ok=True)
t = p.read_text() if p.exists() else ''
if re.search(r'^exclude-newer\s*=', t, re.M):
    t = re.sub(r'^exclude-newer\s*=.*$', 'exclude-newer = \"7 days\"', t, flags=re.M)
else:
    t = (t.rstrip() + '\n' if t.strip() else '') + 'exclude-newer = \"7 days\"\n'
p.write_text(t)
"
```
